activerecord/lib/active_record/reflection.rb

```diff
@@ -581,14 +581,11 @@ def add_as_through(seed)
         seed + [self]
       end
 
-      protected
-
+      private
         def actual_source_reflection # FIXME: this is a horrible name
```

Comment:
This isn't really private, so the change seems more misleading than explanatory.